py_trees/behaviour.py

```diff
@@ -173,8 +167,19 @@ def tick_once(self):
         for unused in self.tick():
             pass
 
+    def setup_with_descendants(self):
```

Comment:
BTW, I was thinking maybe this should take a visitor so that we can invoke it from BehaviourTree.setup() but never mind.

Reply:
aye, I thought about that too. Still, it's good to keep the scope of this class bounded. Behaviours should be concerned with behaviours, trees with trees, not behaviours with trees.